File: src/main/java/withoutaname/mods/immersivesignals/ImmersiveSignals.java
package withoutaname.mods.immersivesignals;

import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import withoutaname.mods.immersivesignals.modules.signal.SignalRegistration;
import withoutaname.mods.immersivesignals.modules.signalcontroller.SignalControllerRegistration;
import withoutaname.mods.immersivesignals.setup.ModSetup;

// The value here should match an entry in the META-INF/mods.toml file
@Mod(ImmersiveSignals.MODID)
public class ImmersiveSignals {

	public static final String MODID = "immersivesignals";

	private static final Logger LOGGER = LogManager.getLogger();

	public ImmersiveSignals() {
		SignalRegistration.init();
		SignalControllerRegistration.init();

		// Register the setup method for modloading
		FMLJavaModLoadingContext.get().getModEventBus().addListener(ModSetup::init);
	}

}

File: src/main/kotlin/eu/withoutaname/mod/immersivesignals/ImmersiveSignals.kt
package eu.withoutaname.mod.immersivesignals

import eu.withoutaname.mod.immersivesignals.datagen.DataGenerators
import eu.withoutaname.mod.immersivesignals.setup.ClientSetup
import eu.withoutaname.mod.immersivesignals.setup.Registration
import net.minecraftforge.eventbus.api.IEventBus
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

val MOD_BUS: IEventBus
    inline get() = FMLJavaModLoadingContext.get().modEventBus


@Mod(ImmersiveSignals.ID)
class ImmersiveSignals {

    companion object {
        const val ID = "immersivesignals"

        @JvmField
        val logger: Logger = LogManager.getLogger()
    }


    init {
        Registration.init()
        ClientSetup.init()
        DataGenerators.init()
    }
}
